src/util/path.ts

```diff
@@ -1,859 +1,817 @@
 
-    rx = Math.abs(rx);
-    ry = Math.abs(ry);
+  rx = Math.abs(rx);
+  ry = Math.abs(ry);
 
-    var px = -cosTh * toX * 0.5 - sinTh * toY * 0.5,
-        py = -cosTh * toY * 0.5 + sinTh * toX * 0.5,
-        rx2 = rx * rx, ry2 = ry * ry, py2 = py * py, px2 = px * px,
-        pl = rx2 * ry2 - rx2 * py2 - ry2 * px2,
-        root = 0;
+  var px = -cosTh * toX * 0.5 - sinTh * toY * 0.5,
+      py = -cosTh * toY * 0.5 + sinTh * toX * 0.5,
+      rx2 = rx * rx, ry2 = ry * ry, py2 = py * py, px2 = px * px,
+      pl = rx2 * ry2 - rx2 * py2 - ry2 * px2,
+      root = 0;
 
-    if (pl < 0) {
-      var s = Math.sqrt(1 - pl / (rx2 * ry2));
-      rx *= s;
-      ry *= s;
-    }
-    else {
-      root = (large === sweep ? -1.0 : 1.0) *
-              Math.sqrt( pl / (rx2 * py2 + ry2 * px2));
-    }
+  if (pl < 0) {
+    const s = Math.sqrt(1 - pl / (rx2 * ry2));
+    rx *= s;
+    ry *= s;
+  }
+  else {
+    root = (large === sweep ? -1.0 : 1.0) *
+            Math.sqrt( pl / (rx2 * py2 + ry2 * px2));
+  }
 
-    var cx = root * rx * py / ry,
+  const cx = root * rx * py / ry,
         cy = -root * ry * px / rx,
         cx1 = cosTh * cx - sinTh * cy + toX * 0.5,
         cy1 = sinTh * cx + cosTh * cy + toY * 0.5,
         mTheta = calcVectorAngle(1, 0, (px - cx) / rx, (py - cy) / ry),
         dtheta = calcVectorAngle((px - cx) / rx, (py - cy) / ry, (-px - cx) / rx, (-py - cy) / ry);
 
-    if (sweep === 0 && dtheta > 0) {
-      dtheta -= 2 * PI;
-    }
-    else if (sweep === 1 && dtheta < 0) {
-      dtheta += 2 * PI;
-    }
+  if (sweep === 0 && dtheta > 0) {
+    dtheta -= 2 * PI;
+  }
+  else if (sweep === 1 && dtheta < 0) {
+    dtheta += 2 * PI;
+  }
 
-    // Convert into cubic bezier segments <= 90deg
-    var segments = Math.ceil(Math.abs(dtheta / PI * 2)),
-        result = [], mDelta = dtheta / segments,
+  // Convert into cubic bezier segments <= 90deg
+  const segments = Math.ceil(Math.abs(dtheta / PI * 2)),
+        result = new Array(segments), mDelta = dtheta / segments,
         mT = 8 / 3 * Math.sin(mDelta / 4) * Math.sin(mDelta / 4) / Math.sin(mDelta / 2),
         th3 = mTheta + mDelta;
 
-    for (var i = 0; i < segments; i++) {
-      result[i] = segmentToBezier(mTheta, th3, cosTh, sinTh, rx, ry, cx1, cy1, mT, fromX, fromY);
-      fromX = result[i][5];
-      fromY = result[i][6];
-      mTheta = th3;
-      th3 += mDelta;
-    }
-    return result;
+  for (let i = 0; i < segments; i++) {
+    result[i] = segmentToBezier(mTheta, th3, cosTh, sinTh, rx, ry, cx1, cy1, mT, fromX, fromY);
+    fromX = result[i][5];
+    fromY = result[i][6];
+    mTheta = th3;
+    th3 += mDelta;
   }
-
-  /*
-   * Private
-   */
-  function calcVectorAngle(ux, uy, vx, vy) {
-    var ta = Math.atan2(uy, ux),
-        tb = Math.atan2(vy, vx);
-    if (tb >= ta) {
-      return tb - ta;
-    }
-    else {
-      return 2 * Math.PI - (ta - tb);
-    }
+  return result;
+}
+
+/*
+  * Private
+  */
+const calcVectorAngle = (ux, uy, vx, vy) => {
```

Comment:
should be extract to vectors

Reply:
yes i m sure this is duplicated i noticed as well

Reply:
what about this?

Reply:
let's leave this for #8083